src/main/java/org/opentripplanner/routing/algorithm/ItineraryRouters.java

```diff
@@ -0,0 +1,363 @@
+
+        if (!router.graph.transitFeedCovers(request.dateTime)) {
+            throw new RoutingValidationException(List.of(
+                    new RoutingError(RoutingErrorCode.OUTSIDE_SERVICE_PERIOD, InputField.DATE_TIME)
+            ));
+        }
+
+        TransitLayer transitLayer = request.ignoreRealtimeUpdates
+            ? router.graph.getTransitLayer()
+            : router.graph.getRealtimeTransitLayer();
+
+        RaptorRoutingRequestTransitData requestTransitDataProvider = createRequestTransitDataProvider(transitLayer, router.graph);
+
+        this.debugTimingAggregator.finishedPatternFiltering();
+
+        var accessEgresses = getAccessEgresses(router, transitLayer);
+
+        this.debugTimingAggregator.finishedAccessEgress();
+
+        List<Itinerary> itineraries = new ArrayList<>();
+
+        // Prepare transit search
+        RaptorRequest<TripSchedule> raptorRequest = RaptorRequestMapper.mapRequest(
+                request,
+                requestTransitDataProvider.getStartOfTime(),
+                accessEgresses.getAccesses(),
+                accessEgresses.getEgresses()
+        );
+
+        // Route transit
+        RaptorResponse<TripSchedule> transitResponse = raptorService.route(
+            raptorRequest,
+            requestTransitDataProvider
+        );
+
+        this.debugTimingAggregator.finishedRaptorSearch();
+
+        Collection<Path<TripSchedule>> paths = transitResponse.paths();
+
+        if(OTPFeature.OptimizeTransfers.isOn()) {
+            paths = TransferOptimizationServiceConfigurator.createOptimizeTransferService(
+                transitLayer::getStopByIndex,
+                router.graph.getTransferService(),
+                requestTransitDataProvider,
+                raptorRequest,
+                request.transferOptimization
+            ).optimize(transitResponse.paths());
+        }
+
+        // Create itineraries
+
+        RaptorPathToItineraryMapper itineraryMapper = new RaptorPathToItineraryMapper(
+                router.graph,
+                transitLayer,
+                requestTransitDataProvider.getStartOfTime(),
+                request
+        );
+        FareService fareService = router.graph.getService(FareService.class);
+
+        // TODO
+        for (Path<TripSchedule> path : paths) {
+            // Convert the Raptor/Astar paths to OTP API Itineraries
+            Itinerary itinerary = itineraryMapper.createItinerary(path);
+            // Decorate the Itineraries with fare information.
+            // Itinerary and Leg are API model classes, lacking internal object references needed for effective
+            // fare calculation. We derive the fares from the internal Path objects and add them to the itinerary.
+            if (fareService != null) {
+                itinerary.fare = fareService.getCost(path, transitLayer);
+            }
+            itineraries.add(itinerary);
+        }
+
+        checkIfTransitConnectionExists(transitResponse);
```

Comment:
Shouldn't this be checked before, immediately after the call to `RaptorService#route`

Reply:
:+1: , I've moved it there.